discord/message.py

```diff
@@ -291,14 +297,15 @@ class MessageReference:
         .. versionadded:: 1.6
     """
 
-    __slots__ = ('message_id', 'channel_id', 'guild_id', 'resolved', '_state')
+    __slots__ = ('message_id', 'channel_id', 'guild_id', 'fail_if_not_exists', 'resolved', '_state')
 
-    def __init__(self, *, message_id, channel_id, guild_id=None):
+    def __init__(self, *, message_id, channel_id, guild_id=None, fail_if_not_exists=None):
```

Comment:
```suggestion
    def __init__(self, *, message_id, channel_id, guild_id=None, fail_if_not_exists=True):
```

A regular `bool` might be better than an `Optional[bool]`? I'm not sure there's a reason for the tri-bool currently.